### PR TITLE
Migrate to GTest

### PR DESCRIPTION
**Pull request by ciband**
Move from AUnit to googletest

## What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactor
- [ ] Performance
- [x] Tests
- [ ] Build
- [ ] Documentation
- [ ] Code style update
- [ ] Continuous Integration
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Does this PR release a new version?

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch
- [x] All tests are passing
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

**Comment by sleepdefic1t** [comment]
@ciband
I’ve got a working fix for the failing PIO in #121

`utils/json.h`

Remove the line that disables Arduino `String`.

```cpp
#define ARDUINOJSON_ENABLE_ARDUINO_STRING 0 // disable 'String'
```

**Comment by sleepdefic1t** [comment]
Well, that fixed some of it 🤷‍♂️

GTest just won’t make it easy.

**Comment by sleepdefic1t** [comment]
Okay, @ciband 
I think I've got it.

I ran `pio run` from the `test` dir, and saw that `googletest` wasn't actually being saved anywhere..

So, from `test/platformio.ini`, I changed `lib_deps` to explicitly use v1.8.1.
> i.e. `googletest` >> `googletest@1.8.1`

```bash
Memory Usage -> http://bit.ly/pio-memory-usage
DATA:    [=         ]   7.3% (used 23916 bytes from 327680 bytes)
PROGRAM: [======    ]  59.2% (used 776301 bytes from 1310720 bytes)
esptool.py v2.6
================================================================= [SUCCESS] Took 31.54 seconds =================================================================
```

File: test/platformio.ini [hunk 1]
```diff
@@ -15,7 +15,7 @@ build_dir = ../build/.pioenvs
 libdeps_dir = ../extern/.piolibdeps
 
 [common]
-lib_deps = micro-ecc, AUnit, ArduinoJson@6.10.0, BIP66
+lib_deps = micro-ecc, googletest, ArduinoJson@6.10.0, BIP66
```
Comment (sleepdefic1t) [inline test/platformio.ini]: `googletest` >> `googletest@1.8.1` 👍

**Comment by ciband** [comment]
@sleepdefic1t  thanks for the debugging help.  I was going to get to it, you just beat me to it.  Looks like that did it.

**Pull request merged**